### DISCUSSION TITLE
Results from IBM Cobol compiler

### DISCUSSION THREAD

**Discussion by awierzch** [PlummersSoftwareLLC/Primes#368] [Show and tell]
Based on [https://github.com/PlummersSoftwareLLC/Primes/tree/drag-race/PrimeCOBOL/solution_1](url), I run this program on real mainframe. The results:

```bash
Passes: 02355, Time: 50, Avg: 000212 (sec/pass), Limit: 1000000, Count: 0078498,
 Valid: True
fvbakel_Cobol;02355;50;1;algorithm=base,faithful=no,bits=8
```

**Comment by rbergen** [comment]
This is a link to a repo created by @andrew890, that includes the z/OS adaptations for COBOL, C++ and Java solutions from this repo: https://github.com/andrew890/Primes-zOS. I guess the COBOL adaptation is very similar to yours, but I am wondering if it would be possible to run the C++ and Java solutions on a mainframe as well, and post the results those generate?

**Reply by andrew890** [reply]
I had been meaning to publish the results of my tests of C++ and Java on z/OS.

I have written up the results here:
https://www.blackhillsoftware.com/news/2021/08/10/java-vs-c-drag-racing-on-z-os/

The COBOL code is the same as contributed by awierzch. I just simplified the JCL to make it easy to run from the git checkout directory, and tidied up the output formatting a bit. The COBOL score is almost exactly the same.

**Reply by rbergen** [reply]
> I have written up the results here:
> https://www.blackhillsoftware.com/news/2021/08/10/java-vs-c-drag-racing-on-z-os/

That's a very nice post!

**Reply by andrew890** [reply]
Thanks!
